docs/guides/bake/file-definition.md

```diff
@@ -0,0 +1,363 @@
+determined by merging them together.
+
+> **Notes**
+>
+> In the case of compose files, each service corresponds to a target.
+
+A group can specify its list of targets with the `targets` option. A target can
+inherit build options by setting the `inherits` option to the list of targets or
+groups to inherit from.
+
+## HCL definition
+
+HCL definition file is recommended as its experience is more aligned with buildx UX
+and also allows better code reuse, different target groups and extended features.
+
+```hcl
+# docker-bake.hcl
+variable "TAG" {
+  default = "latest"
+}
+
+group "default" {
+  targets = ["db", "webapp-dev"]
+}
+
+target "webapp-dev" {
+  dockerfile = "Dockerfile.webapp"
+  tags = ["docker.io/username/webapp:${TAG}"]
+}
+
+target "webapp-release" {
+  inherits = ["webapp-dev"]
+  platforms = ["linux/amd64", "linux/arm64"]
+}
+
+target "db" {
+  dockerfile = "Dockerfile.db"
+  tags = ["docker.io/username/db"]
+}
+```
+
+Complete list of valid target fields:
+
+* `args`
+* `cache-from`
+* `cache-to`
+* `context`
+* `contexts`
+* `dockerfile`
+* `inherits`
+* `labels`
+* `no-cache`
+* `no-cache-filter`
+* `output`
+* `platform`
+* `pull`
+* `secrets`
+* `ssh`
+* `tags`
+* `target`
+
+## JSON definition
+
+```json
+{
+  "variable": {
+    "TAG": {
+      "default": "latest"
+    }
+  },
+  "group": {
+    "default": {
+      "targets": [
+        "db",
+        "webapp-dev"
+      ]
+    }
+  },
+  "target": {
+    "webapp-dev": {
+      "dockerfile": "Dockerfile.webapp",
+      "tags": [
+        "docker.io/username/webapp:${TAG}"
+      ]
+    },
+    "webapp-release": {
+      "inherits": [
+        "webapp-dev"
+      ],
+      "platforms": [
+        "linux/amd64",
+        "linux/arm64"
+      ]
+    },
+    "db": {
+      "dockerfile": "Dockerfile.db",
+      "tags": [
+        "docker.io/username/db"
+      ]
+    }
+  }
+}
+```
+
+Same list of target fields as [HCL definition](#hcl-definition) are available.
+
+## Compose file
+
+```yaml
+# docker-compose.yml
+services:
+  webapp-dev: &dev
+    build:
+      dockerfile: Dockerfile.webapp
+      tags:
+        - docker.io/username/webapp:latest
+
+  webapp-release:
+    <<: *dev
+    build:
+      x-bake:
+        platforms:
+          - linux/amd64
+          - linux/arm64
+
+  db:
+    image: docker.io/username/db
+    build:
+      dockerfile: Dockerfile.db
+```
+
+> **Limitations**
+>
+> Bake uses the [compose-spec](https://docs.docker.com/compose/compose-file/) to
+> parse a compose file. Some fields are not (yet) available, but you can use
+> the [special extension field `x-bake`](compose-xbake.md).
+>
+> `inherits` service field is also not supported. Use [YAML anchors](https://docs.docker.com/compose/compose-file/#fragments)
+> to reference other services.
+>
+> Specifying variables or global scope attributes is not yet supported for
+> compose files.
+{: .warning }
+
+## Remote definition
+
+You can also use a remote `git` bake definition:
```

Comment:
You can also build bake files directly from a remote Git repository or HTTPS URL.